rfcs/042-editable-programmatic-descriptions.md

```diff
@@ -0,0 +1,100 @@
+
+The same need exists for business specific human generated metadata. This type of data does not exist in a machine generated form that can be easily extracted from a source system. Instead, users need to add/edit this metadata directly in the Amundsen UI. 
+
+## Motivation
+
+This feature allows users to configure their Amundsen installation to support one or more editable programmatic descriptions. These editable programmatic descriptions allow users to enrich the metadata displayed on the table details page with additional business specific human generated information. This feature will allow them to add/edit these programmatic descriptions directly in the UI.
+
+## Guide-level Explanation (aka Product Details)
+
+An editable programmatic description is the same as a regular programmatic description, except that it is maintained directly in the UI by a human instead of by a scheduled databuilder job. For example, users could configure an editable programmatic description with a source of data_usage_policy and then update the data_usage_policy on individual table detail pages with a link to the applicable data usage policy on their company's content management site.
+
+Similar to table descriptions, if a table is disabled for editing, then so are the editable programmatic descriptions.
+
+Databuilder jobs may still be useful on a one-off (i.e. non-scheduled basis) to seed Amundsen with manually curated data or perform one-off data conversations.
+
+There will be no impact to existing programmatic descriptions. They will remain non-editable via the UI unless users intentionally update the existing configuration in their Amundsen installation to make them editable.
+
+## UI/UX-level Explanation
+
+Editable programmatic descriptions would be displayed alongside programmatic descriptions in the same way they are displayed today. They will be rendered as markdown, located in one of three sections (left, right, other) in the page and sorted according to display_order.
+
+There are two differences between editable and non-editable programmatic descriptions:
+1. They are editable (if the table is editable)
+2. They will always be displayed. If there is not a value then placeholder text will be displayed instead.
+
+_**Note:** The pink text is only visible in the mocks to call out design details and changes._
+![Table Detail Page](../assets/042/table-detail-page.png)
+
+## Reference-level Explanation (aka Technical Details)
+
+The implementation will require:
+- Updates to the frontend and metadata service to support editing programmatic descriptions via the API
+- Updates to the frontend to display editable programmatic descriptions in the table details page 
+
+### Metadata
+
+- Add an optional "description_source" parameter to the existing put_table_description API endpoint that defaults to None.
```

Comment:
Is `description_source` being used to determine whether or not a prog desc can be edited? If so, what are some potential values here and is there a need to have a well-defined, limited set of consistent values?

Reply:
The `description_source` would be used to populate the `source` attribute of [ProgrammaticDescription](https://github.com/amundsen-io/amundsen/blob/45f77100cd13c4faa4f922e1daf5e2e92fd23434/common/amundsen_common/models/table.py#L118). Similar to how databuilder currently works for loading descriptions, if the `description_source` is None (the default value) then the `description` populate the table's Description, otherwise the `description` and `description_source` populate a Programmatic_Description. I opted for `description_source` since it looks like we already have a parameter called `source` that is used to denote where the source of the update (see [here](https://github.com/amundsen-io/amundsen/blob/45f77100cd13c4faa4f922e1daf5e2e92fd23434/frontend/amundsen_application/api/metadata/v0.py#L311) and [here](https://github.com/amundsen-io/amundsen/blob/45f77100cd13c4faa4f922e1daf5e2e92fd23434/frontend/amundsen_application/static/js/ducks/tableMetadata/api/v0.ts#L103)).